deny.toml

```diff
@@ -0,0 +1,10 @@
+[licenses]
+version = 2
+allow = ["Apache-2.0", "BSD-3-Clause", "ISC", "MIT", "MPL-2.0", "OpenSSL", "Unicode-3.0"]
```

Comment:
I think we [need one more here](https://github.com/rustls/rustls/actions/runs/14237900888/job/39900863213?pr=2412) for Zlib:

```
error[rejected]: failed to satisfy license requirements
  ┌─ registry+https://github.com/rust-lang/crates.io-index#zlib-rs@0.5.0:4:12
  │
4 │ license = "Zlib"
  │            ━━━━
  │            │
  │            license expression retrieved via Cargo.toml `license`
  │            rejected: license is not explicitly allowed
  ```